test/testuninitvar.cpp

```diff
@@ -6394,6 +6394,16 @@ class TestUninitVar : public TestFixture {
                         "    return x[0];\n"
                         "}\n");
         ASSERT_EQUALS("", errout.str());
+
+        valueFlowUninit("void f(bool* p) {\n" // #12287
+                        "    if (p)\n"
+                        "        *p = true; \n"
+                        "}\n"
+                        "void g() {\n"
+                        "    bool b;\n"
+                        "    f(&b);\n"
+                        "}\n");
+        ASSERT_EQUALS("", errout.str());
```

Comment:
There should be a valueflow test to make sure the valueflow for `p` in `*p = true` is not set to an uninitialized value.